runtime/src/main/scala/com/soundcloud/twinagle/ProtoService.scala

```diff
@@ -1,24 +1,32 @@
 package com.soundcloud.twinagle
 
-import com.twitter.finagle.Service
+import com.twitter.finagle.{Filter, Service}
 import com.twitter.finagle.http.{Request, Response}
 import com.twitter.util.Future
 import scalapb.{GeneratedMessage, GeneratedMessageCompanion}
 
-case class ProtoService(rpcs: Seq[ProtoRpc]) {
-  assert(rpcs.map(_.metadata.service).toSet.size == 1, "inconsistent services in metadata")
-}
+case class ProtoService(rpcs: Seq[ProtoRpcBuilder])
+
 object ProtoService {
   implicit val asProtoService: AsProtoService[ProtoService] = (t: ProtoService) => t
 }
 
 case class ProtoRpc(metadata: EndpointMetadata, svc: Service[Request, Response])
-object ProtoRpc {
```

Comment:
Please note that the `Proto*` classes are public and thus part of the API of the library. These changes would break backwards compat and force a 2.0 release. It wouldn't be the end of the world, but I'd like to see if we can avoid this so soon after the 1.0 release.

How about, instead of introducing a new `ProtoRpcBuilder` type, we could keep around the existing `ProtoRpc.apply()` method, delegating to a new method in `object ProtoRpc` with three parameters that builds the final `ProtoRpc`. Does that make sense?

Reply:
> delegating to a new method in object ProtoRpc with three parameters that builds the final ProtoRpc. Does that make sense?

I tried to do this in the begging, but I couldn't figure our how can we reused the existing method. The problem is that we don't have access to the message filters when we call it here: https://github.com/soundcloud/twinagle/blob/master/codegen/src/main/scala/com/soundcloud/twinagle/codegen/TwinagleServicePrinter.scala#L34